### DISCUSSION TITLE
Dimension of each kernel attribute in Multitask GP model

### DISCUSSION THREAD

**Discussion by GHU2021** [cornellius-gp/gpytorch#2021] [Q&A] [answered]
Hi,

When I implemented Multitask GP Regression (MTGP) with RQ Kernel and for example 10 tasks, I noticed that the MTGP has 10 raw_var values but only one raw_lenghscale and one  raw_alpha value. 
I mean model.covar_module.data_covar_module.raw_alpha is a torch.Size([1]), model.covar_module.data_covar_module.raw_lengthscale is a torch.Size([1, 1]), 
and model. covar_module.task_covar_module.raw_var is a torch.Size([24]).

I defined MultitaskGPModel as follows:
class MultitaskGPModel(gpytorch.models.ExactGP):
    def __init__(self, train_x, train_y, likelihood):
        super(MultitaskGPModel, self).__init__(train_x, train_y, likelihood)
        self.mean_module = gpytorch.means.MultitaskMean(
            gpytorch.means.ConstantMean(), num_tasks=10)
        self.covar_module = gpytorch.kernels.MultitaskKernel(
            gpytorch.kernels.RQKernel(), num_tasks=10)
    def forward(self, x):
        mean_x = self.mean_module(x)
        covar_x = self.covar_module(x)
        return gpytorch.distributions.MultitaskMultivariateNormal(mean_x, covar_x)

Also, I defined likelihood and model as follows:
likelihood = gpytorch.likelihoods.MultitaskGaussianLikelihood(num_tasks=10)
model = MultitaskGPModel(train_x, train_y, likelihood)

Is this the limitation of MTGP in GPyTorch?
If not, How can I revise MultitaskGPModel, likelihood, and/or model so that raw_alpha and raw_lenghscale have 10 elements each?

Thank you for your help.

**Comment by gpleiss** [comment, marked as answer]
The GPyTorch MultitaskKernel implements the "Intrinsic Coregionalization Model" (see, e.g., [Alvarez et al., 2011](https://arxiv.org/pdf/1106.6251.pdf)), which essentially corresponds to each task sharing the same kernel hyperparameters.

You want the "Linear Model of Coregionalization" (see same paper), where each task is a linear combination of GPs that could use different kernels. There is a [LCMKernel in GPyTorch](https://docs.gpytorch.ai/en/stable/kernels.html#lcmkernel).

**Reply by GHU2021** [reply]
Dear Dr. Pleiss,

Thank you for quick response and helpful information.
